base/loading.jl

```diff
@@ -315,21 +316,78 @@ all platforms, including those with case-insensitive filesystems like macOS and
 Windows.
 """
 function require(mod::Symbol)
-    _require(mod)
-    # After successfully loading, notify downstream consumers
-    if toplevel_load[] && myid() == 1 && nprocs() > 1
-        # broadcast top-level import/using from node 1 (only)
-        @sync for p in procs()
-            p == 1 && continue
-            @async remotecall_wait(p) do
-                if !isbindingresolved(Main, mod) || !isdefined(Main, mod)
-                    _require(mod)
+    if !root_module_exists(mod)
+        _require(mod)
+        # After successfully loading, notify downstream consumers
+        if toplevel_load[] && myid() == 1 && nprocs() > 1
+            # broadcast top-level import/using from node 1 (only)
+            @sync for p in procs()
+                p == 1 && continue
+                @async remotecall_wait(p) do
+                    require(mod)
+                    nothing
                 end
             end
         end
+        for callback in package_callbacks
+            invokelatest(callback, mod)
+        end
     end
-    for callback in package_callbacks
-        invokelatest(callback, mod)
+    return root_module(mod)
+end
+
+const loaded_modules = ObjectIdDict()
+const module_keys = ObjectIdDict()
+
+function register_root_module(key, m::Module)
+    if haskey(loaded_modules, key)
+        oldm = loaded_modules[key]
+        if oldm !== m
+            name = module_name(oldm)
+            warn("replacing module $name.")
```

Comment:
Are we doing Uppercase messages? I don't precisely remember what style we've started to settle on.

Reply:
We don't seem to be totally consistent yet. On a quick search it's close to 50/50. There are some other occurrences of this same message that are lowercase.